hrjobcontract/js/test/controllers/form/form-general_test.js

```diff
@@ -0,0 +1,89 @@
+define([
+  'common/lodash',
+  'common/moment',
+  'common/angularMocks',
+  'common/mocks/services/hr-settings-mock',
+  'job-contract/app'
+], function (_, moment) {
+  'use strict';
+
+  describe('FormGeneralCtrl', function () {
+    var ctrl, $controller, $provide, $rootScope, $scope;
+
+    beforeEach(function () {
+      module('hrjc', 'common.mocks', function (_$provide_) {
+          $provide = _$provide_;
+      });
+      inject(['HR_settingsMock', function (HR_settingsMock) {
+          $provide.value('HR_settings', HR_settingsMock);
+      }]);
+    });
+
+    beforeEach(inject(function (_$controller_, _$rootScope_) {
+      $controller = _$controller_;
+      $rootScope = _$rootScope_;
+    }));
+
+    describe('Min and max allowed dates', function () {
+      describe('when there are no period dates already set', function () {
```

Comment:
The `describe` block gives a lexical scope to a set of tests, whether it's one or multiple. Notice that it allows me to have control over the "before" condition for each set of tests, something that I could not do without it.

Also it gives better structure in terms of readability of the test suite, making it more english-like